packages/java/src/common/java-common.ts

```diff
@@ -0,0 +1,36 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/**
+ * The WS endpoint path to the Java service.
+ */
+export const JavaServicePath = '/services/java';
+
+/**
+ * JavaService symbol for DI.
+ */
+export const JavaService = Symbol('JavaService');
+
+/**
+ * This service provides functionality to get all needed information for initializing Java Language Server.
+ */
+export interface JavaService {
+    /**
+     * Finds all java extensions which are provided in extensions.
+     * @return an array of paths to the plugins
+     */
+    collectJavaExtensions(): Promise<string[] | undefined>;
```

Comment:
What is the purpose to return `undefined` ? It is better to return an empty array if there are no extensions.

Reply:
"collectExtensionBundleIds()" perhaps? Or is it the path to the bundle jar? A more specific name would help.

Reply:
It is a path to the bundle jar